src/twisted/web/client.py

```diff
@@ -1158,7 +1158,10 @@ def stopProducing(self):
         stopping the underlying L{CooperativeTask}.
         """
         self._inputFile.close()
-        self._task.stop()
+        try:
+            self._task.stop()
+        except task.TaskFinished:
```

Comment:
My preference would be for a fix that properly tracked the state of `self._task` and didn't provoke this exception in the first place.

It seems to me that the right place to close `_inputFile` would be in `maybeStopped` (and also the currently no-op callback there), which is *already* checking for `TaskStopped`; it's just not doing anything with it.

Don't quite have time for a full review right now though, and I might just land this as-is to quash the error.  But if you'd like to do something that actually tracked the completed state of `self._task` so that `stopProducing` could do the right thing I know I'd be happier about it :)

Reply:
Thanks for the feedback!

I'm certainly not opposed to it, and to honest, one goal proposing this (the simplest thing that I felt _could_ work) was to elicit opinions from those who know the code.  

It'll take me a bit (doesn't seem like there's a huge rush, and I'm not anxious to land it for landing its sake), but what I'm thinking is I'll leave this as-is, for reference, then do another PR with an alternative in line with your suggestion.  I'll reference the PRs to each other in the ticket as well as on Github.  Then at least there will be a couple options.

I'm not opposed to doing it to this PR, but I thought it might be helpful to show the genesis of the next solution.

Reply:
Sounds good to me!  Hopefully one day I won't be exhausted and sick all the time and I can confidently stamp this :)

Reply:
No doubt, best wishes for health and energy!